storage/provider/loop.go

```diff
@@ -66,16 +90,69 @@ func (lvs *loopVolumeSource) rootDeviceDir() string {
 	return filepath.Join(dirParts...)
 }
 
-func (lvs *loopVolumeSource) CreateVolumes(params []storage.VolumeParams) ([]storage.BlockDevice, error) {
-	panic("not implemented")
+func (lvs *loopVolumeSource) CreateVolumes(args []storage.VolumeParams) ([]storage.BlockDevice, error) {
+
+	blockDevices := make([]storage.BlockDevice, len(args))
+	for idx, arg := range args {
+		id := providerId(arg)
+		filePath := filepath.Join(lvs.rootDeviceDir(), id)
+		if err := createBlockFile(lvs.runCmd, filePath, arg.Size); err != nil {
+			return nil, err
+		}
+
+		deviceName, err := attachToBlockDevice(lvs.runCmd, filePath)
+
+		// If we ran out of loop devices, create another.
+		if err == NoLoopDeviceErr {
+			nextDevNum, err := findAvailableDeviceNumber(lvs.runCmd)
+			if err == nil {
+				createLoopDevice(lvs.runCmd, nextDevNum)
```

Comment:
- should be checking error here.
- in what situation do you need to create loop devices?